### PR TITLE
fix Issue 12832 - asm movdqu accepts wrong operand size

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by WalterBright** [comment]
Blocked by https://github.com/dlang/druntime/pull/3194

There may be more.

**Comment by Geod24** [comment]
Test?

**Comment by WalterBright** [comment]
Currently blocked by https://github.com/dlang/phobos/pull/7597

File: src/dmd/iasmdmd.d [hunk 1]
```diff
@@ -817,24 +817,25 @@ RETRY:
         }
         case 2:
         {
-            //printf("opflags1 = "); asm_output_flags(opflags[0]); printf(" ");
-            //printf("opflags2 = "); asm_output_flags(opflags[1]); printf("\n");
+            enum log = false;
+            if (log) { printf("opflags1 = "); asm_output_flags(opflags[0]); printf("\n"); }
```
Comment (NilsLankila) [inline src/dmd/iasmdmd.d:821]: `static if` and the next 6 times too.

Comment (WalterBright) [reply src/dmd/iasmdmd.d:821]: In this case, `if` does the same thing and is shorter, too, as the line is getting a bit long. The code for the printf isn't generated.

**Comment by WalterBright** [comment]
What's this from semaphoreci ?
```
+ generated/linux/release/64/dmd -version=NoBackend -version=GC -version=NoMain -Jgenerated/dub -Jsrc/dmd/res -Isrc -i -run test/dub_package/frontend.d
Error: program killed by signal 11
```

**Comment by NilsLankila** [comment]
No idea, I got the same crash for #11573 too. That PR was so simple that the failure is definitively from the side of the service not DMD, otherwise it would manifest in the Auto-Tester and circle-ci as well.

Note that you can try the same command localy (after a `$ dub build`) and verify that it works, as I just did.

```bash
$ generated/linux/release/64/dmd -version=NoBackend -version=GC -version=NoMain -Jgenerated/dub -Jsrc/dmd/res -Isrc -i -run test/dub_package/frontend.d
$ echo $?
0
```

**Comment by MoonlightSentinel** [comment]
> What's this from semaphoreci ?

That's unrelated, currently debugging in #11593 

> No idea, I got the same crash for #11573 too. That PR was so simple that the failure is definitively from the side of the service not DMD, otherwise it would manifest in the Auto-Tester and circle-ci as well.

Neither of those runs the dub package test. I guess it's related to the previous `-lowmem` issue because the tests runs the frontend with enabled GC.

**Comment by NilsLankila** [comment]
> ??:? void etc.linux.memoryerror.sigsegvUserspaceProcess(void*) [0x801f53]
??:? void etc.linux.memoryerror.sigsegvDataHandler() [0x801e3a]
??:? _ZN12TypeFunction9isnothrowEb [0x721065]
??:? _ZN12TypeFunction10syntaxCopyEv [0x71f291]

I have touched to TypeFunction.syntaxcopy quite recently. #11461. Could it be the culprit ? Gotta check that.

**Comment by WalterBright** [comment]
@NilsLankila @MoonlightSentinel thanks for the information and work to track down that problem. Since it is unrelated to this PR, can we get this one pulled?

**Review by ghost — APPROVED** [review]

**Comment by NilsLankila** [comment]
auto-merge does not seem to work when there's a failed check. You could also accept and merge the fix for semaphore ci (#11594) and rebase this PR after.

**Comment by NilsLankila** [comment]
Auto-merge toggled on

**Comment by WalterBright** [comment]
Thanks, guys!

**Pull request merged**